slack-lint-checks/src/main/java/slack/lint/denylistedapis/DenyListedApiDetector.kt

```diff
@@ -246,6 +246,20 @@ internal class DenyListedApiDetector : Detector(), SourceCodeScanner, XmlScanner
           ),
         arguments = listOf("*"),
       ),
+      DenyListedEntry(
+        className = "java.util.Date",
+        functionName = MatchAll,
+        fieldName = MatchAll,
+        errorMessage =
+          "Use java.time.Instant or java.time.ZonedDateTime instead. There is no reason to use java.util.Date in Java 8+."
+      ),
+      DenyListedEntry(
+        className = "java.text.DateFormat",
```

Comment:
What are your thoughts on `android.text.format.DateFormat`?

Reply:
Also I'd recommend we include the common `java.text.SimpleDateFormat` as well.

Reply:
First one's a tricky one, because I know there's some actual string localization there that doesn't really have an alternative (like human-readable dates). Maybe we do denylist it, but in slack we build a java.time.* abstraction over it that folks use.

Agreed on SimpleDateFormat 👍